dag_in_context/src/dag2svg.rs

```diff
@@ -3,24 +3,69 @@ use std::{
     rc::Rc,
 };
 
-use dot_structures::{Edge, EdgeTy, Graph, Id, Node, NodeId, Stmt, Subgraph, Vertex};
+use dot_structures::{Attribute, Edge, EdgeTy, Graph, Id, Node, NodeId, Stmt, Subgraph, Vertex};
 
-use crate::schema::{Expr, RcExpr, TreeProgram};
+use crate::schema::{Constant, Expr, RcExpr, TreeProgram};
+
+// We need one node for every unique expression, scope pair
+// Expressions may be shared across scopes but should be shown multiple times
+#[derive(Eq, PartialEq, Hash)]
+struct UniqueExpr {
+    expr: *const Expr,
+    scope: *const Expr,
+}
```

Comment:
Why are you using `*const Expr` instead of `&Expr`? This is very odd, isn't it?

Reply:
added comment